burnman/layer.py

```diff
@@ -0,0 +1,797 @@
+        pressure = np.ravel(
+            odeint((lambda p, x: gfunc(x) * rhofunc(x)), pressure_top, depth))
+
+        return pressure
+
+    @property
+    def mass(self):
+        """
+        Calculates the mass of the layer [kg]
+        """
+        mass = 0.0
+        radii = self.radii[::-1]
+        density = self.density[::-1]
+        rhofunc = UnivariateSpline(radii, density)
+        mass = np.abs(quad(lambda r: 4 * np.pi * rhofunc(r) *
+                           r * r, radii[0], radii[-1])[0])
+        return mass
+
+    @property
+    def moment_of_inertia(self):
+        """
+        Returns the moment of inertia of the layer [kg m^2]
+        """
+        moment = 0.0
+        radii = self.radii[::-1]
+        density = self.density[::-1]
+        rhofunc = UnivariateSpline(radii, density)
+        moment = np.abs(quad(lambda r: 8.0 / 3.0 * np.pi * rhofunc(r)
+                             * r * r * r * r, radii[0], radii[-1])[0])
+        return moment
+
+    @property
+    def gravity(self):
+        """
+        Returns gravity of the layer [m s^(-2)]
+        """
+        return self._compute_gravity(self.density, self.gravity_bottom)
+
+    @property
+    def bullen(self):
+        """
+        Returns the Bullen parameter
+        """
+        kappa = self.bulk_sound_velocity * self.bulk_sound_velocity * self.density
+        phi = self.bulk_sound_velocity * self.bulk_sound_velocity
+        dkappadP = np.gradient(kappa, edge_order=2) / \
+            np.gradient(self.pressure, edge_order=2)
+        dphidz = np.gradient(phi,
+                             edge_order=2) / np.gradient(self.depths,
+                                                         edge_order=2) / self.gravity
+        bullen = dkappadP - dphidz
+        return bullen
+
+    @property
+    def brunt_vasala(self):
+        kappa = self.bulk_sound_velocity * self.bulk_sound_velocity * self.density
+        brunt_vasala = self.density * self.gravity * \
+            self.gravity * (self.bullen - 1.) / kappa
+        return brunt_vasala
+
+    @property
+    def pressure(self):
+        """
+        Returns current pressure that was set with :func:`~burnman.material.Material.set_state`.
+
+
+        Notes
+        -----
+        - Aliased with :func:`~burnman.material.Material.P`.
+
+        Returns
+        -------
+        pressure : array of floats
+            Pressure in [Pa].
+        """
+        return self._pressures
+
+    @property
+    def temperature(self):
+        """
+        Returns current temperature that was set with :func:`~burnman.material.Material.set_state`.
+
+        Notes
+        -----
+        - Aliased with :func:`~burnman.material.Material.T`.
+
+        Returns
+        -------
+        temperature : array of floats
+            Temperature in [K].
+        """
+        return self._temperatures
+
+    @material_property
+    def internal_energy(self):
+        """
+        Returns the internal energy of the layer.
```

Comment:
Do you mean the internal energy at each pressure-temperature point? Again, this isn't clear.